### PR TITLE
feat: add annotation api

### PR DESCRIPTION
**Pull request by voicis**
#506 

- Added new `fiori-annotation-api` package.
- Removed ESLint rule that was conflicting with prettier.

### CODE REVIEW

File: packages/fiori-annotation-api/Readme.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- Serialize annotations to XML or SAP CAP CDS.
+
+
+## Limitations
+
+The following features are not supported:
+- Using changes with logical and arithmetic operators: `Apply`, `And`, `Add` etc.
+
+## Usage
+See usage in `test/fiori-service.test`.
```
Comment (Klaus-Keller) [inline packages/fiori-annotation-api/Readme.md:31]: File does not exist, but there is `test/unit/fiori-service.test.ts`

Comment (voicis) [reply packages/fiori-annotation-api/Readme.md]: Fixed in https://github.com/SAP/open-ux-tools/pull/2025/commits/1eb7b33a22bdd8c0cc98c3ed3070870012bbbdea

**Review by Jimmy-Joseph19 — DISMISSED** [review]
Checked the code, built it locally, but did not try testing consuming this version. 
Test coverage looks ok
LGTM, 
Approving.

**Review by Jimmy-Joseph19 — APPROVED** [review]
Approving for the new reafactoring commits. 
DId not test locally.

**Review by Klaus-Keller — APPROVED** [review]
Thanks @voicis, I've checked 
- the overall package structure
- changeset
- code coverage
- Sonar code quality 

Review of implementation details needs to be done by annotation experts.

Approved from my side

**Pull request merged**